app/models/script.rb

```diff
@@ -0,0 +1,24 @@
+class Script < ApplicationRecord
+
+  has_many :events
+
+  before_validation :add_or_update_checksum
+
+  validates_presence_of :text
+  validate :unique_text
+
+  private
+
+  def add_or_update_checksum
+    return nil unless self.text # avoid triggering error on Script.new.valid?
+    checksum = Digest::SHA1.hexdigest(self.text)
+    if self.checksum.nil? || self.checksum != checksum
+      self.checksum = checksum
+    end
+  end
```

Comment:
The goal to minimize the cost of the project in the long run.

The more duped data in the DB, the bigger the DB.  Bigger the DB, the higher cost.

Hashing the submitted script isn't perfect solution, but it's a start.  May be best to hash just the non-whitespace.  

You mentioned alternative ways to quickly verify that we're not bloating the DB with duplicate records.  What did you have in mind?